scylla-proxy/src/proxy.rs

```diff
@@ -633,12 +633,25 @@ impl Doorkeeper {
         let (tx_driver, rx_driver) = mpsc::unbounded_channel::<ResponseFrame>();
         let event_register_flag = Arc::new(AtomicBool::new(false));
 
-        tokio::task::spawn(new_worker().receiver_from_driver(driver_read, tx_request));
+        let (
+            compression_writer_request_processor,
+            compression_reader_receiver_from_driver,
+            compression_reader_receiver_from_cluster,
+            compression_reader_sender_to_driver,
+            compression_reader_sender_to_cluster,
+        ) = compression::make_compression_infra();
+
+        tokio::task::spawn(new_worker().receiver_from_driver(
+            driver_read,
+            tx_request,
+            compression_reader_receiver_from_driver,
+        ));
```

Comment:
🤔 This is not the first time we need to share some state between the tasks - we already share event_register_flag. I think it is probable that we will need more stuff in the future, for example if we decide to add more advanced Conditions (like those matching the query string of prepared statement).

Maybe we should have some `ConnectionState` and/or  `ProxyState` for the shared state? They would be shared and could contain `OnceLock<Option<Compression>>`, atomic bool for events etc. It would make it easier to add new such items in the future.

Reply:
Makes sense. This is however out of scope of this PR. I can try to refactor this according to your suggestion in a follow-up.

Reply:
Please do so, or at least open an issue about it.